ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -353,8 +353,12 @@ void fillChunks() {
       if (chunkFillingCompletedSuccessfully) {
         PutChunk lastChunk = getBuildingChunk();
         if (lastChunk != null) {
-          lastChunk.onFillComplete(true);
-          updateChunkFillerWaitTimeMetrics();
+          if (chunkCounter != 0 && lastChunk.buf.position() == 0) {
```

Comment:
`lastChunk.buf.position() == 0` is not a reliable check since buffers can have position > 0 and yet have no data because position == limit.

Also, what happens if there are zero sized buffers in the middle? I think it should be fine - but a test would be useful (if not already there)